server/sonar-server/src/main/java/org/sonar/server/measure/custom/ws/SearchAction.java

```diff
@@ -114,10 +115,10 @@ public void handle(Request request, Response response) throws Exception {
   }
 
   @CheckForNull
-  private Long searchLastSnapshot(DbSession dbSession, ComponentDto project) {
-    SnapshotDto lastSnapshot = dbClient.snapshotDao().selectLastSnapshotByComponentUuid(dbSession, project.uuid());
+  private Long searchLastSnapshotDate(DbSession dbSession, ComponentDto project) {
+    Optional<SnapshotDto> lastSnapshot = dbClient.snapshotDao().selectLastSnapshotByComponentUuid(dbSession, project.projectUuid());
```

Comment:
it really should be `project.uuid()` unless the component can be something else than a project, in which case variables are poorly named